plugins/retrolab/fps_retrolab/routes.py

```diff
@@ -34,141 +47,211 @@
     name="themes",
 )
 
+for path1 in (retrolab_dir / "labextension" / "static").glob("remoteEntry.*.js"):
+    load = f"static/{path1.name}"
+    break
+federated_extensions = [
+    {
+        "extension": "./extension",
+        "load": load,
+        "name": "@retrolab/lab-extension",
```

Comment:
Probably we already have logic for handling the federated extension in the `jupyterlab` plugin?

RetroLab uses the same set of federated extension as JupyterLab so I'm wondering it it's possible to share the same logic? Or maybe it's better to keep them separate for now since `fps-jupyterlab` and `fps-retrolab` are two different plugins?

Reply:
Yes, I was thinking about sharing some logic between both plugins, but since they are currently two distinct packages, it requires having a common package for this. As it's not much logic, maybe we can live with a bit of code duplication for now?

Reply:
Right it would be fine to have the same logic duplicated for now.

For reference with the current tornado based apps, that logic is implemented in `jupyterlab_server` which is used by both JupyterLab and RetroLab (so the logic lives in one place).